controller/action/ContentActions.class.php

```diff
@@ -364,4 +364,32 @@ public static function preparePostListPartial(array $vars): array
       'posts' => array_slice(Post::find(static::VIEW_FOLDER_NEWS, Post::SORT_DATE_DESC), 0, $count)
     ];
   }
+  public static function preparePostCategoryFilter(string $category)
+  {
+    Response::enableHttpCache();
+
+    $filter_post = [];
+
+    $posts = array_filter(
+      Post::find(static::VIEW_FOLDER_NEWS, Post::SORT_DATE_DESC),
+      function(Post $post) {
+        return !$post->getDate() || $post->getDate()->format('U') <= date('U');
```

Comment:
`return (($post->getCategory() === $category) && (!$post->getDate() || $post->getDate()->format('U') <= date('U')));`

Reply:
(You'll also need to say `use $category`)